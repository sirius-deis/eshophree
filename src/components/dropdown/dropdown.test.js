import { render, screen, fireEvent } from '@testing-library/react';

import Dropdown from './dropdown';

describe('Dropdown component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Dropdown>Some content</Dropdown>);
    expect(container).toMatchSnapshot();
  });
  it('should render element', () => {
    render(<Dropdown>Some content</Dropdown>);
    const content = screen.getByText(/Some content/);
    expect(content).toBeInTheDocument();
  });
  it('should fire an click event', () => {
    const fn = jest.fn();
    render(<Dropdown onMouseLeaveHandler={fn}>Some content</Dropdown>);
    fireEvent.mouseLeave(screen.getByText('Some content'));
    expect(fn).toHaveBeenCalled();
  });
});
