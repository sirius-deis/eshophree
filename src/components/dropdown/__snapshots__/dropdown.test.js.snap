// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should match snapshot 1`] = `
<div>
  <div
    class="sc-bgqQcB CeERh"
  >
    Some content
  </div>
</div>
`;
